arch/sim/src/sim/posix/sim_host_v4l2.c

```diff
@@ -140,7 +142,7 @@ int host_video_dqbuf(struct host_video_dev_s *vdev, uint8_t *addr,
 
           default:
             perror("VIDIOC_DQBUF");
-            return -errno;
+            return host_errno_convert(-errcode);
```

Comment:
need return errno from host_errno_convert with irq_disable to avoid losing errno